clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -484,4 +484,18 @@ public void testPartitionsForWithNullTopic() {
             // expected
         }
     }
+
+    @Test
+    public void testBatchExpiryMs() {
+        Properties props = new Properties();
+        props.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:2002");
+        props.setProperty(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, "5");
+        KafkaProducer<String, String> producerWithDefaultBatchExpiry = new KafkaProducer<>(props, new StringSerializer(), new StringSerializer());
+
+        assertEquals(5, producerWithDefaultBatchExpiry.batchExpiryMs());
+        props.setProperty(ProducerConfig.BATCH_EXPIRY_MS, "10");
+
+        KafkaProducer<String, String> producerWithConfiguredBatchExpiry = new KafkaProducer<>(props, new StringSerializer(), new StringSerializer());
+        assertEquals(10L, producerWithConfiguredBatchExpiry.batchExpiryMs());
```

Comment:
We should close the producers after usage (maybe use try with resources).

Reply:
ack. Updated the patch.